packages/vm/src/vm.ts

```diff
@@ -117,11 +121,13 @@ export class VM {
       ) {
         enableProfiler = true
       }
+      const evmOpts = opts.evmOpts ?? {}
       opts.evm = await EVM.create({
-        common: opts.common,
-        stateManager: opts.stateManager,
-        blockchain: opts.blockchain,
-        profiler: {
+        ...evmOpts,
+        common: opts.evmOpts?.common ?? opts.common,
+        stateManager: opts.evmOpts?.stateManager ?? opts.stateManager,
+        blockchain: opts.evmOpts?.blockchain ?? opts.blockchain,
+        profiler: opts.evmOpts?.profiler ?? {
```

Comment:
Doesn't this:

```typescript
        common: opts.common,
        stateManager: opts.stateManager,
        blockchain: opts.blockchain,
        profiler: {
          enabled: enableProfiler,
        },
        ...evmOpts
```

Achieve the same? Or did I miss some side effects? This will first setup the object with the `opts` settings and then override (if those exist) in `evmOpts`?

Reply:
(this is a nit, I agree the current state works as well)

Reply:
Needed to have a really careful look on this to be sure, but: yes, I think this works, nice optimization! 🙂 

Have updated!